### PR TITLE
bruno: 1.11.0 -> 1.12.2

### LINKED ISSUE #298288: Update request: bruno 1.11.0 → 1.12.2
- Package name: `bruno`
- Latest released version: `1.12.2`

- Current version on the unstable channel: `1.11.0`
- Current version on the stable/release channel: `1.1.1`

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
- @water-sucks @lucasew @kashw2 @mattpolzin

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mattpolzin**
## Description of changes

I needed to patch the lockfile to get the new `@tailwindcss/forms` dependency in there prior to doing the usual fixup and then building. I created the patch by simply running `npm install` from a fresh checkout of the Bruno project's repo.

There is a closed PR upstream already to update the lock file in this manner.

Closes https://github.com/NixOS/nixpkgs/issues/298288.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 298809` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bruno</li>
  </ul>
</details>

File: pkgs/by-name/br/bruno/package.nix
Comment (drupol) [inline pkgs/by-name/br/bruno/package.nix:39]: Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

**Review by drupol — DISMISSED** [review]

File: pkgs/by-name/br/bruno/package.nix
Comment (lucasew) [reply pkgs/by-name/br/bruno/package.nix:39]: I'd remove that patch. `npm-lockfile-fix` should handle this issue.

Comment (mattpolzin) [reply pkgs/by-name/br/bruno/package.nix:39]: It didn't in this case. I don't know much about `npm-lockfile-fix`, but I've tried this same build with and without the patch; the build fails with the classic package not cached error without the path. It's the same error you can see in the logs from the auto-update attempt by r-ryantm.

Comment (mattpolzin) [reply pkgs/by-name/br/bruno/package.nix:39]: I'll fix this patch to match nixpkgs conventions but then we can decide if we want to merge this PR or not; the choices being: merge this PR which updates to 1.12.2 but guarantees we will need to manually remove the patch after the next version is released or alternatively holding off on updating Bruno in nixpkgs in hopes that 1.12.3 follows quickly with no need for the patch.

Comment (drupol) [reply pkgs/by-name/br/bruno/package.nix:39]: Just add a comment on top of the patch explaining that this patch is only for version 1.12.2 and that it needs to be removed at the next point release and that will be enough.

Comment (drupol) [reply pkgs/by-name/br/bruno/package.nix]: Fine fine.

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 298809` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bruno</li>
  </ul>
</details>

**Pull request merged**
